chain/network/src/routing.rs

```diff
@@ -262,6 +262,10 @@ pub struct RoutingTable {
     pub peer_forwarding: HashMap<PeerId, Vec<PeerId>>,
     /// Store last update for known edges.
     pub edges_info: HashMap<(PeerId, PeerId), Edge>,
+    /// Shared version of edges_info used by multiple threads
+    edges_info_shared: Arc<Mutex<HashMap<(PeerId, PeerId), u64>>>,
+    /// List of edges that should be added
+    edges_to_add_shared: Arc<Mutex<Vec<Edge>>>,
```

Comment:
I would like to use a Lock-free unbounded multiple writers one reader queue, for `edges_to_add_shared`, similar to one that `Actix` uses. I couldn't find one on crates.io, but it's there probably somewhere.

Reply:
Otherwise, we may block `PeerManager` while doing `edges_to_add_shared.lock()`.

Reply:
Though I think `Mutex` is fine. We are only going to add a small number of edges in total.